modules/apps/roles/roles-admin-test/src/testIntegration/java/com/liferay/roles/admin/internal/segments/entry/test/RoleSegmentsEntryRetrieverTest.java

```diff
@@ -0,0 +1,140 @@
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.roles.admin.internal.segments.entry.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.portal.kernel.model.Role;
+import com.liferay.portal.kernel.model.role.RoleConstants;
+import com.liferay.portal.kernel.search.BaseModelSearchResult;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.RoleTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.roles.admin.segments.entry.RoleSegmentsEntryManager;
+import com.liferay.roles.admin.segments.entry.RoleSegmentsEntryRetriever;
+import com.liferay.segments.model.SegmentsEntry;
+import com.liferay.segments.test.util.SegmentsTestUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Drew Brokke
+ */
+@RunWith(Arquillian.class)
+public class RoleSegmentsEntryRetrieverTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext());
+	}
+
+	@After
+	public void tearDown() {
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testComponentRegistered() {
+		Assert.assertNotNull(_roleSegmentsEntryRetriever);
+	}
+
+	@Test
+	public void testSearch() throws Exception {
+		_segmentsEntries.add(
+			SegmentsTestUtil.addSegmentsEntry(TestPropsValues.getGroupId()));
+
+		_assertSearch(null, 0);
+
+		_addRoleSegmentsEntry();
+
+		_assertSearch(null, 1);
+	}
+
+	@Test
+	public void testSearchByKeyword() throws Exception {
+		String keywords = RandomTestUtil.randomString();
+
+		_segmentsEntries.add(
+			SegmentsTestUtil.addSegmentsEntry(
+				TestPropsValues.getGroupId(), keywords + 1));
+
+		_assertSearch(keywords, 0);
+
+		_addRoleSegmentsEntry();
+
+		_assertSearch(keywords, 0);
+
+		_addRoleSegmentsEntry(keywords + 2);
```

Comment:
if you add a role segments entry with just `keywords` here, does it pass the test? It could be that the fuzzy search is not fully implemented.